arkouda/dataframe.py

```diff
@@ -1525,12 +1525,43 @@ def shape(self):
         (3, 2)
         """
         self.update_nrows()
-        num_cols = len(self._columns)
+        num_cols = len(self._column_names)
         nrows = self._nrows
         return (nrows, num_cols)
 
     @property
     def columns(self):
+        """
+        An Index where the values are the column names of the dataframe.
+
+        Returns
+        -------
+        arkouda.index.Index
+            The values of the index are the column names of the dataframe.
+
+        Examples
+        --------
+
+        >>> import arkouda as ak
+        >>> ak.connect()
+        >>> df = ak.DataFrame({'col1': [1, 2], 'col2': [3, 4]})
+        >>> df
+
+        +----+--------+--------+
+        |    |   col1 |   col2 |
+        +====+========+========+
+        |  0 |      1 |      3 |
+        +----+--------+--------+
+        |  1 |      2 |      4 |
+        +----+--------+--------+
+
+        >>> df.columns
+        Index(array(['col1', 'col2']), dtype='<U0')
+        """
+        return Index(self._column_names)
+
+    @property
+    def column_names(self):
```

Comment:
is `column_names` a property of a pandas dataframe?

Reply:
No, it would be a new feature.